roles/lib_yaml_editor/library/yedit.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+'''
+module for openshift cloud secrets
+'''
+
+import os
+import yaml
+
+class YeditException(Exception):
+    pass
+
+class Yedit(object):
+    ''' Class to modify yaml files '''
+
+    def __init__(self, filename):
+        self.filename = filename
+        self.__yaml_dict = None
+        self.get()
+
+    @property
+    def yaml_dict(self):
+        return self.__yaml_dict
+
+    @yaml_dict.setter
+    def yaml_dict(self, value):
+        ''' setter method for yaml_dict '''
+        self.__yaml_dict = value
+
+    @staticmethod
+    def remove_entry(data, keys):
+        ''' remove an item from a dictionary with key notation a.b.c
+            d = {'a': {'b': 'c'}}}
+            keys = a.b
+            item = c
+        '''
+        if "." in keys:
+            key, rest = keys.split(".", 1)
+            if key in data.keys():
+                Yedit.remove_entry(data[key], rest)
+        else:
+            del data[keys]
+
+    @staticmethod
+    def add_entry(data, keys, item):
+        ''' Add an item to a dictionary with key notation a.b.c
+            d = {'a': {'b': 'c'}}}
+            keys = a.b
+            item = c
+        '''
+        if "." in keys:
+            key, rest = keys.split(".", 1)
+            if key not in data:
+                data[key] = {}
+
+            if not isinstance(data, dict):
+                raise YeditException('Invalid add_entry called on a [%s] of type [%s].' % (data, type(data))))
+            else:
+                Yedit.add_entry(data[key], rest, item)
+
+        else:
+            data[keys] = item
+
+
+    @staticmethod
+    def get_entry(data, keys):
+        ''' Get an item from a dictionary with key notation a.b.c
+            d = {'a': {'b': 'c'}}}
+            keys = a.b
+            return c
+        '''
+        if keys and "." in keys:
+            key, rest = keys.split(".", 1)
+            if not isinstance(data[key], dict):
+                raise YeditException('Invalid get_entry called on a [%s] of type [%s].' % (data, type(data)))
+
+            else:
+                return Yedit.get_entry(data[key], rest)
+
+        else:
+            return data.get(keys, None)
+
+
+    def write(self):
+        ''' write to file '''
+        with open(self.filename, 'w') as yfd:
+            yfd.write(yaml.dump(self.yaml_dict, default_flow_style=False))
```

Comment:
Below it says that I'm able to specify "json" as the file type.

But here it always only writes yaml.

Is that on purpose?